data_cleaner.do

```diff
@@ -0,0 +1,274 @@
+/* City Name Cleaner File HUD Replication */
+/* Written by: Matthew D. Webb, Areez Gangji and Anthony McCanny  */
+/* Date: September 4, 2023  */
+
+// Section 1: Clean up zip codes and merge data
+	// Clean up zip codes
+	gen lenzip = strlen(hzip) // Generate length of zip code
+	gen zip = substr(hzip,1,5) // Clean up 9/10 digit zips
+	destring zip, replace force // Convert string to numeric to match other file
+
+	// Merge city data
+	merge m:1 zip using "${DATA}/zipinfo.dta" // Merge with zip data
+	drop if _merge==2 // Drop unmatched entries from zip file
+	drop _merge
+	drop allaccept17-allaccept31 // Drop unneeded alternatives
+
+	// Merge county data
+	merge m:1 zip using "${DATA}/zipinfo-county.dta" // Merge with county data matched to zip code
+	drop if _merge==2 // Drop unmatched entries from zip file
+	drop _merge
+
+	// Rename hcityx column to hcity to standardize across different data files
+	capture rename hcityx hcity
+
+	// Generate city name flag
+	gen good_city = 0 // Initialize flag for whether city name is good
+
+// Section 2: Clean up the strings
+	// Generate uppercase values of hcity and primary_city
+	// These will be used later to match on subsections of the string
+	gen upper_hcity = upper(hcity)
+    gen upper_primary = upper(primary_city)
+
+	// Create a new function to clean city names
+	// Drop the existing program 'clean_city' if it exists to avoid conflicts
+	cap program drop clean_city
+	
+	// Define a new program 'clean_city'
+	program define clean_city
+		args varname newvarname
+        gen `newvarname' = `varname'
+		
+		// Loop over a set of characters
+		foreach char in " " "." "-" "'" "," "0" "1" "2" "3" "4" "5" "6" "7" "8" "9" "\\" {
```

Comment:
might be unnecessary ( i imagine the effects would be limited) but do we want to check for a few more characters? e.g. 
`"`, `:` `#`, `$` `%`, `[`, `]`, `(`, `)`.

Reply:
Made the change! I don't think it changed anything, but it doesn't hurt to be complete!